### PR TITLE
correct checking `args` and cleanup in `mixins/enumerable`

### PR DESCRIPTION
**Pull request by bekzod**

### CODE REVIEW

File: packages/ember-runtime/lib/mixins/enumerable.js [hunk 1]
```diff
@@ -741,7 +741,7 @@ const Enumerable = Mixin.create({
       let method = x && x[methodName];
 
       if ('function' === typeof method) {
-        ret[idx] = args ? method.apply(x, args) : x[methodName]();
```
Comment (bekzod) [inline packages/ember-runtime/lib/mixins/enumerable.js]: here `args` is always array

**Comment by stefanpenner** [comment]
awesome. Geeze that sauce test is driving us crazy. @bekzod do you have any free cycles to investigate whats going on there? It seems the stability of our sauce tests have gone down recently.

**Comment by bekzod** [comment]
@stefanpenner I will have a look, sure

File: packages/ember-utils/lib/invoke.js [hunk 2]
```diff
@@ -44,6 +44,6 @@ export function canInvoke(obj, methodName) {
 */
 export function tryInvoke(obj, methodName, args) {
   if (canInvoke(obj, methodName)) {
-    return args ? applyStr(obj, methodName, args) : applyStr(obj, methodName);
+    return applyStr(obj, methodName, args);
```
Comment (bekzod) [inline packages/ember-utils/lib/invoke.js]: there is already `args` existence check inside  `applyStr`

File: packages/ember-runtime/lib/mixins/array.js [hunk 3]
```diff
@@ -159,7 +154,7 @@ export function arrayContentDidChange(array, startIdx, removeAmt, addAmt) {
 const EMBER_ARRAY = symbol('EMBER_ARRAY');
 
 export function isEmberArray(obj) {
-  return obj && !!obj[EMBER_ARRAY];
+  return obj && obj[EMBER_ARRAY];
```
Comment (bekzod) [inline packages/ember-runtime/lib/mixins/array.js]: [EMBER_ARRAY] === true

File: packages/ember-runtime/lib/mixins/array.js [hunk 4]
```diff
@@ -57,11 +56,7 @@ export function removeArrayObserver(array, target, opts) {
 }
 
 export function objectAt(content, idx) {
-  if (content.objectAt) {
-    return content.objectAt(idx);
-  }
-
-  return content[idx];
+  return content.objectAt ? content.objectAt(idx) : content[idx];
```
Comment (krisselden) [inline packages/ember-runtime/lib/mixins/array.js]: typeof content.objectAt === 'function'

**Review by krisselden — CHANGES_REQUESTED** [review]
looks good

File: packages/ember-runtime/lib/mixins/enumerable.js [hunk 5]
```diff
@@ -741,7 +741,7 @@ const Enumerable = Mixin.create({
...
+        ret[idx] = args.length ? method.apply(x, args) : x[methodName]();
```
Comment (bekzod) [inline packages/ember-runtime/lib/mixins/enumerable.js]: @krisselden maybe it should be just `method.apply(x, args)` without args check ?

File: packages/ember-runtime/lib/mixins/enumerable.js (hunk 1 above)
Comment (bekzod) [inline packages/ember-runtime/lib/mixins/enumerable.js]: maybe just default to `method.apply(x, args)` always here ?

**Review by krisselden — APPROVED** [review]

**Pull request merged**